packages/vats/test/test-boot.js

```diff
@@ -37,45 +37,47 @@ const vatRoots = {
 
 const noop = () => {};
 
-const makeMock = t => ({
-  devices: {
-    command: /** @type { any } */ ({ registerInboundHandler: noop }),
-    mailbox: /** @type { any } */ ({
-      registerInboundHandler: noop,
-    }),
-    timer: /** @type { any } */ ({}),
-    plugin: /** @type { any } */ ({ registerReceiver: noop }),
-    ...devices,
-  },
-  vats: {
-    vattp: /** @type { any } */ (
-      Far('vattp', {
-        registerMailboxDevice: noop,
-        addRemote: () => ({}),
-      })
-    ),
-    comms: Far('comms', {
-      addRemote: noop,
-      addEgress: noop,
-      addIngress: async () => ({
-        getConfiguration: () => ({ _: 'client configuration' }),
+const makeMock = t =>
+  harden({
```

Comment:
With the inappropriate `harden` in track-turns added back in at https://github.com/Agoric/agoric-sdk/pull/5892/commits/346a60a6c8cda47962e8b81b428e02bd45d09f31 , the `harden`s added to `test-boot` are no longer strictly needed into for the test to run successfully. But their absence is a bug that the `harden` in trackTurns was masking. Going through a phase in which that `harden` was removed revealed these bugs. I want to include them in this PR because this PR also puts that `harden` back in, for compatibility, but at the cost of continuing to mask bugs like this.

Curiously, these missing hardens was the *only* bug revealed during the phase in which the patched trackTurns did not harden. With only these changes, we were green under CI prior to https://github.com/Agoric/agoric-sdk/pull/5892/commits/346a60a6c8cda47962e8b81b428e02bd45d09f31